From 7e6c1238e19835663105c95fb0e38f52d4b283a5 Mon Sep 17 00:00:00 2001
From: Andoni Morales Alastruey <ylatuya@gmail.com>
Date: Sun, 14 Feb 2010 20:55:42 +0100
Subject: [PATCH 3/3] [dshowvideosink] Implement XOverlay expose()

---
 sys/dshowvideosink/dshowvideosink.cpp |   13 +++++++++++++
 1 files changed, 13 insertions(+), 0 deletions(-)

diff --git a/sys/dshowvideosink/dshowvideosink.cpp b/sys/dshowvideosink/dshowvideosink.cpp
index 1355b2b..a717422 100644
--- a/sys/dshowvideosink/dshowvideosink.cpp
+++ b/sys/dshowvideosink/dshowvideosink.cpp
@@ -90,6 +90,7 @@ static GstFlowReturn gst_dshowvideosink_show_frame (GstVideoSink *sink, GstBuffe
 
 /* GstXOverlay methods */
 static void gst_dshowvideosink_set_window_id (GstXOverlay * overlay, ULONG window_id);
+static void gst_dshowvideosink_expose (GstXOverlay * overlay);
 
 /* TODO: event, preroll, buffer_alloc? 
  * buffer_alloc won't generally be all that useful because the renderers require a 
@@ -115,6 +116,7 @@ static void
 gst_dshowvideosink_xoverlay_interface_init (GstXOverlayClass * iface)
 {
   iface->set_xwindow_id = gst_dshowvideosink_set_window_id;
+  iface->expose = gst_dshowvideosink_expose;
 }
 
 static void
@@ -1406,6 +1408,17 @@ gst_dshowvideosink_show_frame (GstVideoSink *vsink, GstBuffer *buffer)
   return ret;
 }
 
+static void
+gst_dshowvideosink_expose(GstXOverlay * overlay)
+{
+  GstDshowVideoSink *sink = GST_DSHOWVIDEOSINK (overlay);
+  GstBuffer *last_buffer;
+
+  last_buffer = gst_base_sink_get_last_buffer(GST_BASE_SINK(sink));
+  if (last_buffer != NULL)
+    gst_dshowvideosink_show_frame(GST_VIDEO_SINK(sink), last_buffer);
+}
+
 /* TODO: How can we implement these? Figure that out... */
 static gboolean
 gst_dshowvideosink_unlock (GstBaseSink * bsink)
-- 
1.6.4.msysgit.0

