Index: sys/dshowvideosink/dshowvideosink.cpp
===================================================================
--- sys/dshowvideosink/dshowvideosink.cpp	(revision 639)
+++ sys/dshowvideosink/dshowvideosink.cpp	(working copy)
@@ -225,6 +225,7 @@
 {
   gst_dshowvideosink_clear (sink);
 
+  sink->graph_lock = g_mutex_new();
   sink->com_lock = g_mutex_new();
   sink->com_initialized = g_cond_new();
   sink->com_uninitialize = g_cond_new();
@@ -258,6 +259,8 @@
     while (sink->comInitialized);
   }
 
+  g_mutex_free (sink->graph_lock);
+
   G_OBJECT_CLASS (parent_class)->finalize (gobject);
 }
 
@@ -933,16 +936,20 @@
 
   switch (transition) {
     case GST_STATE_CHANGE_PLAYING_TO_PAUSED:
+      GST_DSHOWVIDEOSINK_GRAPH_LOCK(sink);
       rettmp = gst_dshowvideosink_pause_graph (sink);
       if (rettmp == GST_STATE_CHANGE_FAILURE)
         ret = rettmp;
       sink->graph_running = FALSE;
+      GST_DSHOWVIDEOSINK_GRAPH_UNLOCK(sink);
       break;
     case GST_STATE_CHANGE_PAUSED_TO_READY:
+      GST_DSHOWVIDEOSINK_GRAPH_LOCK(sink);
       rettmp = gst_dshowvideosink_stop_graph (sink);
       if (rettmp == GST_STATE_CHANGE_FAILURE)
         ret = rettmp;
       sink->graph_running = FALSE;
+      GST_DSHOWVIDEOSINK_GRAPH_UNLOCK(sink);
       break;
     case GST_STATE_CHANGE_READY_TO_NULL:
       gst_dshowvideosink_clear (sink);
@@ -1435,6 +1442,7 @@
   }
 
   GST_DEBUG_OBJECT (sink, "Pushing buffer through fakesrc->renderer");
+  GST_DSHOWVIDEOSINK_GRAPH_LOCK(sink);
   if (!sink->graph_running){
     retst = gst_dshowvideosink_start_graph(sink);
     if (retst == GST_STATE_CHANGE_FAILURE)
@@ -1446,6 +1454,7 @@
     if (retst == GST_STATE_CHANGE_FAILURE)
       return GST_FLOW_WRONG_STATE;
   }
+  GST_DSHOWVIDEOSINK_GRAPH_UNLOCK(sink);
   GST_DEBUG_OBJECT (sink, "Done pushing buffer through fakesrc->renderer: %s", gst_flow_get_name(ret));
 
   return ret;
Index: sys/dshowvideosink/dshowvideosink.h
===================================================================
--- sys/dshowvideosink/dshowvideosink.h	(revision 639)
+++ sys/dshowvideosink/dshowvideosink.h	(working copy)
@@ -41,6 +41,9 @@
 typedef struct _GstDshowVideoSink GstDshowVideoSink;
 typedef struct _GstDshowVideoSinkClass GstDshowVideoSinkClass;
 
+#define GST_DSHOWVIDEOSINK_GRAPH_LOCK(sink)	g_mutex_lock (GST_DSHOWVIDEOSINK (sink)->graph_lock)
+#define GST_DSHOWVIDEOSINK_GRAPH_UNLOCK(clock) g_mutex_unlock (GST_DSHOWVIDEOSINK (sink)->graph_lock)
+
 /* Renderer-specific support classes */
 class RendererSupport
 {
@@ -99,6 +102,9 @@
   /* If we use an app-supplied window, we need to hook its WNDPROC */
   WNDPROC prevWndProc;
 
+  /* Lock for transitions */
+  GMutex *graph_lock;
+
   gboolean comInitialized;
   GMutex   *com_lock;
   GCond    *com_initialized;
