--- ext/x264/gstx264enc.c	Mon Jul 19 17:33:28 2010
+++ ext/x264/gstx264enc.c	Mon Jul 19 17:31:59 2010
@@ -559,6 +559,10 @@
   encoder->x264param.p_log_private = encoder;
   encoder->x264param.i_log_level = X264_LOG_DEBUG;
 
+  /* setup appropriate timebase for gstreamer */
+  encoder->x264param.i_timebase_num = 1;
+  encoder->x264param.i_timebase_den = 1000000000;
+
   gst_x264_enc_reset (encoder);
 }
 
