Index: win32/common/config.h
===================================================================
--- win32/common/config.h	(revision 480)
+++ win32/common/config.h	(working copy)
@@ -75,7 +75,7 @@
 #undef HAVE_CFPREFERENCESCOPYAPPVALUE
 
 /* Define to 1 if you have the `cosh' function. */
-#undef HAVE_COSH
+#define HAVE_COSH
 
 /* Define if the host CPU is an Alpha */
 #undef HAVE_CPU_ALPHA
@@ -271,7 +271,7 @@
 #undef HAVE_SHOUT2
 
 /* Define to 1 if you have the `sinh' function. */
-#undef HAVE_SINH
+#define HAVE_SINH
 
 /* Define to enable soup http client plugin (2.4) (used by souphttpsrc). */
 #undef HAVE_SOUP
