Index: win32/common/config.h
===================================================================
--- win32/common/config.h	(revision 535)
+++ win32/common/config.h	(working copy)
@@ -30,9 +30,6 @@
 /* gettext package name */
 #define GETTEXT_PACKAGE "gst-plugins-ugly-0.10"
 
-/* macro to use to show function name */
-#undef GST_FUNCTION
-
 /* Defined if gcov is enabled to force a rebuild due to config.h changing */
 #undef GST_GCOV_ENABLED
 
