Index: gst/h264parse/gsth264parse.c
===================================================================
--- gst/h264parse/gsth264parse.c	(revision 562)
+++ gst/h264parse/gsth264parse.c	(working copy)
@@ -966,11 +966,13 @@
   for (i = 0; i < MAX_SPS_COUNT; i++) {
     if (h264parse->sps_buffers[i])
       g_slice_free (GstH264Sps, h264parse->sps_buffers[i]);
+    h264parse->sps_buffers[i] = NULL;
     gst_buffer_replace (&h264parse->sps_nals[i], NULL);
   }
   for (i = 0; i < MAX_PPS_COUNT; i++) {
     if (h264parse->pps_buffers[i])
       g_slice_free (GstH264Pps, h264parse->pps_buffers[i]);
+    h264parse->pps_buffers[i] = NULL;
     gst_buffer_replace (&h264parse->pps_nals[i], NULL);
   }
   h264parse->sps = NULL;
@@ -996,10 +998,7 @@
   h264parse->last_outbuf_dts = GST_CLOCK_TIME_NONE;
 
   list = h264parse->codec_nals;
-  while (list) {
-    gst_buffer_unref (list->data);
-    list = g_slist_next (list);
-  }
+  g_slist_foreach (list, (GFunc) gst_buffer_unref, NULL);
   g_slist_free (h264parse->codec_nals);
   h264parse->codec_nals = NULL;
 
