Index: ext/mpeg2dec/gstmpeg2dec.c
===================================================================
--- ext/mpeg2dec/gstmpeg2dec.c	(revision 510)
+++ ext/mpeg2dec/gstmpeg2dec.c	(working copy)
@@ -972,11 +972,11 @@
       GST_TIME_FORMAT,
       (picture->flags & PIC_FLAG_PROGRESSIVE_FRAME ? "prog" : "    "),
       (picture->flags & PIC_FLAG_TOP_FIELD_FIRST ? "tff" : "   "),
-#if MPEG2_RELEASE >= MPEG2_VERSION(0,5,0)
+//#if MPEG2_RELEASE >= MPEG2_VERSION(0,5,0)
       (picture->flags & PIC_FLAG_REPEAT_FIRST_FIELD ? "rff" : "   "),
-#else
+/*#else
       "unknown rff",
-#endif
+#endif*/
       (picture->flags & PIC_FLAG_SKIP ? "skip" : "    "),
       (picture->flags & PIC_FLAG_COMPOSITE_DISPLAY ? "composite" : "         "),
       picture->nb_fields, GST_BUFFER_OFFSET (outbuf),
