Index: gst/gstplugin.c
===================================================================
--- gst/gstplugin.c	(revision 712)
+++ gst/gstplugin.c	(working copy)
@@ -670,7 +670,11 @@
   GModule *module;
   gboolean ret;
   gpointer ptr;
+#if GLIB_CHECK_VERSION(2,25,0)
+  GStatBuf file_status;
+#else
   struct stat file_status;
+#endif
   GstRegistry *registry;
   gboolean new_plugin = TRUE;
 
@@ -1466,7 +1470,13 @@
 }
 
 static guint
-gst_plugin_ext_dep_get_hash_from_stat_entry (struct stat *s)
+gst_plugin_ext_dep_get_hash_from_stat_entry (
+#if GLIB_CHECK_VERSION(2,25,0)
+    GStatBuf *s
+#else
+    struct stat *s
+#endif
+)
 {
   if (!(s->st_mode & (S_IFDIR | S_IFREG)))
     return (guint) - 1;
@@ -1523,7 +1533,11 @@
    * the same order, and not in a random order */
   while ((entry = g_dir_read_name (dir))) {
     gboolean have_match;
+#if GLIB_CHECK_VERSION(2,25,0)
+    GStatBuf s;
+#else
     struct stat s;
+#endif
     gchar *full_path;
     guint fhash;
 
@@ -1580,7 +1594,12 @@
    * and going through each entry to see if it matches one of our filenames. */
   if (!recurse_into_dirs && !partial_names) {
     for (i = 0; filenames[i] != NULL; ++i) {
+#if GLIB_CHECK_VERSION(2,25,0)
+      GStatBuf s;
+#else
       struct stat s;
+#endif
+
       gchar *full_path;
       guint fhash;
 
Index: gst/gstregistry.c
===================================================================
--- gst/gstregistry.c	(revision 712)
+++ gst/gstregistry.c	(working copy)
@@ -1106,7 +1106,11 @@
     return FALSE;
 
   while ((dirent = g_dir_read_name (dir))) {
+#if GLIB_CHECK_VERSION(2,25,0)
+    GStatBuf file_status;
+#else
     struct stat file_status;
+#endif
 
     filename = g_build_filename (path, dirent, NULL);
     if (g_stat (filename, &file_status) < 0) {
Index: gst/gstregistrybinary.c
===================================================================
--- gst/gstregistrybinary.c	(revision 712)
+++ gst/gstregistrybinary.c	(working copy)
@@ -363,7 +363,11 @@
       continue;
 
     if (plugin->flags & GST_PLUGIN_FLAG_CACHED) {
+#if GLIB_CHECK_VERSION(2,25,0)
+      GStatBuf statbuf;
+#else
       struct stat statbuf;
+#endif
 
       if (g_stat (plugin->filename, &statbuf) < 0 ||
           plugin->file_mtime != statbuf.st_mtime ||
