Index: gst-plugins-base/ext/ogg/gstoggstream.c
===================================================================
--- gst-plugins-base/ext/ogg/gstoggstream.c	(revision 479)
+++ gst-plugins-base/ext/ogg/gstoggstream.c	(working copy)
@@ -73,7 +73,7 @@
   GstOggMapPacketDurationFunc packet_duration_func;
 };
 
-static const GstOggMap mappers[];
+static const GstOggMap mappers[18];
 
 GstClockTime
 gst_ogg_stream_get_packet_start_time (GstOggStream * pad, ogg_packet * packet)
