--- sys/directdraw/gstdirectdrawsink.c	Thu Jul  8 10:27:45 2010
+++ sys/directdraw/gstdirectdrawsink.c	Thu Jul 15 16:24:09 2010
@@ -880,7 +880,7 @@
   }
 
   if (buf == NULL) {
-    GST_ERROR_OBJECT (ddrawsink, "No buffer to render.");
+    /*GST_ERROR_OBJECT (ddrawsink, "No buffer to render.");*/
     return GST_FLOW_ERROR;
   } else if (!ddrawsink->video_window) {
     GST_WARNING_OBJECT (ddrawsink, "No video window to render to.");
