Index: gst/typefind/gsttypefindfunctions.c
===================================================================
--- gst/typefind/gsttypefindfunctions.c	(revision 479)
+++ gst/typefind/gsttypefindfunctions.c	(working copy)
@@ -33,6 +33,10 @@
 #define USE_GIO
 #endif
 
+#ifdef _MSC_VER
+#define strncasecmp strnicmp
+#endif
+
 #include <gst/gsttypefind.h>
 #include <gst/gstelement.h>
 #include <gst/gstversion.h>
