Index: gst/udp/gstudpnetutils.c
===================================================================
--- gst/udp/gstudpnetutils.c	(revision 491)
+++ gst/udp/gstudpnetutils.c	(working copy)
@@ -166,6 +166,11 @@
 
   socklen = sizeof (addr);
   if ((ret = getsockname (sockfd, (struct sockaddr *) &addr, &socklen)) < 0) {
+#ifdef G_OS_WIN32
+  /* This error is returned on windows if the socket is not bound */
+  if (WSAGetLastError() == WSAEINVAL)
+	  ret = 0;
+#endif
     return ret;
   }
 
