Index: ext/speex/gstspeexdec.c
===================================================================
--- ext/speex/gstspeexdec.c	(revision 526)
+++ ext/speex/gstspeexdec.c	(working copy)
@@ -538,7 +538,7 @@
     dec->callback.callback_id = SPEEX_INBAND_STEREO;
     dec->callback.func = speex_std_stereo_request_handler;
     dec->callback.data = &dec->stereo;
-    dec->stereo = (SpeexStereoState) SPEEX_STEREO_STATE_INIT;
+    dec->stereo = *(speex_stereo_state_init());
     speex_decoder_ctl (dec->state, SPEEX_SET_HANDLER, &dec->callback);
   }
 
