--- bin/winvlc.c	Fri Jun 25 14:36:52 2010
+++ bin/winvlc.c	Fri Jun 25 14:36:48 2010
@@ -46,11 +46,7 @@
 static void check_crashdump(void);
 LONG WINAPI vlc_exception_filter(struct _EXCEPTION_POINTERS *lpExceptionInfo);
 # endif
-typedef enum _HEAP_INFORMATION_CLASS {
-        HeapCompatibilityInformation,
-        HeapEnableTerminationOnCorruption
-} HEAP_INFORMATION_CLASS;
-WINBASEAPI BOOL WINAPI HeapSetInformation(HANDLE,HEAP_INFORMATION_CLASS,PVOID,SIZE_T);
+
 #define HeapEnableTerminationOnCorruption (HEAP_INFORMATION_CLASS)1
 #endif
 
