--- gio/gnullapplication.c	Fri Jun 18 13:43:46 2010
+++ gio/gnullapplication.c	Fri Jun 18 13:41:37 2010
@@ -50,13 +50,13 @@
 static void
 _g_application_platform_remote_invoke_action (GApplication  *app,
                                               const gchar   *action,
-                                              guint          timestamp)
+                                              GVariant      *platform_data)
 {
 }
 
 static void
 _g_application_platform_remote_quit (GApplication *app,
-                                     guint         timestamp)
+                                     GVariant      *platform_data)
 {
 }
 
