--- mingw/x86_64-w64-mingw32/include/iphlpapi.h	Fri Jun 25 03:19:03 2010
+++ mingw/x86_64-w64-mingw32/include/iphlpapi.h	Fri Jun 25 03:19:00 2010
@@ -14,7 +14,7 @@
 #include <ipexport.h>
 #include <iptypes.h>
 
-#if (_WIN32_WINNT == 0x502 && WINVER == 0x502)
+//#if (_WIN32_WINNT == 0x502 && WINVER == 0x502)
 typedef enum _TCP_TABLE_CLASS {
   TCP_TABLE_BASIC_LISTENER,
   TCP_TABLE_BASIC_CONNECTIONS,
@@ -26,7 +26,7 @@
   TCP_TABLE_OWNER_MODULE_CONNECTIONS,
   TCP_TABLE_OWNER_MODULE_ALL
 } TCP_TABLE_CLASS, *PTCP_TABLE_CLASS;
-#endif
+//#endif
 
   DWORD WINAPI GetNumberOfInterfaces(PDWORD pdwNumIf);
   DWORD WINAPI GetIfEntry(PMIB_IFROW pIfRow);
